lib/internal/http2/core.js

```diff
@@ -2680,7 +2678,6 @@ class Http2SecureServer extends TLSServer {
     options = initializeTLSOptions(options);
     super(options, connectionListener);
     this[kOptions] = options;
-    this.timeout = kDefaultSocketTimeout;
```

Comment:
```suggestion
    this.timeout = 0;
```

...or if `kDefaultSocketTimeout` is set to `0`, leave the line as-is?